insights/parsers/__init__.py

```diff
@@ -329,6 +329,19 @@ def keyword_search(rows, **kwargs):
     Takes a list of dictionaries and finds all the dictionaries where the
     keys and values match those found in the keyword arguments.
 
+    Keys in the row data have ' ' and '-' replaced with '_', so they can
+    match the keyword argument parsing.  For example, the keyword argument
+    'fix_up_path' will match a key named 'fix-up path'.
+
+    In addition, several suffixes can be added to the key name to do partial
+    matching of values:
+
+    * '__contains' will test whether the data value contains the given
+      value.
+    * '__startswith' tests if the data value starts with the given value
+    * '__lower_value' compares the lower-case version of the data and given
+      values.
+
```

Comment:
This is useful and needed functionality, but I think there are better ways to implement this added functionality. 

If there are users asking for this functionality, and this is how they wanted it implemented, or if there are other reasons for doing it this way, then the following is not so important.  But if not, Paul would you consider changing the way you did this to something along the lines ....

First the spaces and dashes in keys part of this PR.  Python already has a syntax that allows one to put spaces and dashes and anything else one might need into mapping keys.  

Instead of:
    result = keyword_search(rows, key_with_spaces='value')

just do:
   result = keyword_search(rows, **{ 'key with spaces' : 'value' })

Yes, it a few extra characters over keyword arguments, but it doesn't require any extra code in keyword_search, and will never conflate spaces, dashes, or underscores.

Second, the value match functions, '__contains', etc.  I think it would be better to allow boolean functions for the values to match would be better than a fixed set of suffixes on the keys.  

Instead of:
    result = keyword_search(rows, item__contains='c')

do:
    result = keyword_search(rows, item=lambda v: 'c' in v)

To implement this, you would change keyword_search to look at the type of the value in each kwarg key/value pair.  If it's a string compare it equality, if it's a function (a callable) then call it.